client_handlers.go

```diff
@@ -76,14 +94,20 @@ func clientHandshakeHandler(c *Conn) error {
 
 		case *handshakeMessageServerHello:
 			for _, extension := range h.extensions {
-				if e, ok := extension.(*extensionUseSRTP); ok {
+				switch e := extension.(type) {
+				case *extensionUseSRTP:
 					profile, ok := findMatchingSRTPProfile(e.protectionProfiles, c.localSRTPProtectionProfiles)
 					if !ok {
 						return fmt.Errorf("Server responded with SRTP Profile we do not support")
 					}
 					c.state.srtpProtectionProfile = profile
+				case *extensionUseExtendedMasterSecret:
+					c.state.extendedMasterSecret = true
 				}
 			}
+			if c.requireExtendedMasterSecret && !c.state.extendedMasterSecret {
+				return fmt.Errorf("Client required Extended Master Secret extension, but server does not support it")
```

Comment:
Would it be worth offering some internal switch so we can disable EMS? It would be nice to assert that these error conditions are hit/returned.

Reply:
Yes that's a good idea, maybe a config option so the user can disable it if wanted?

Reply:
I switched it to an enum with three options:
Request (default)
Required
Disabled

Reply:
And tests added!